agent/registration/host_registrar.go

```diff
@@ -129,3 +141,37 @@ func (hr *HostRegistrar) GetNetworkStatus() []infrastructurev1beta1.NetworkStatu
 	}
 	return Network
 }
+
+// getHostInfo gets the host platform details...
+func (hr *HostRegistrar) getHostInfo() (infrastructurev1beta1.HostInfo, error) {
+	hostInfo := infrastructurev1beta1.HostInfo{}
+
+	hostInfo.Architecture = runtime.GOARCH
+	hostInfo.OSName = runtime.GOOS
+
+	if distribution, err := getOperatingSystem(ioutil.ReadFile); err != nil {
+		return hostInfo, errors.Wrap(err, "failed to get host operating system image")
+	} else {
+		hostInfo.OSImage = distribution
+	}
+	return hostInfo, nil
+}
+
+// getOperatingSystem gets the name of the current operating system image.
+func getOperatingSystem(f func(string) ([]byte, error)) (string, error) {
+	rex := regexp.MustCompile("(PRETTY_NAME)=(.*)")
+
+	bytes, err := f("/etc/os-release")
+	if err != nil && os.IsNotExist(err) {
+		// /usr/lib/os-release in stateless systems like Clear Linux
+		bytes, err = f("/usr/lib/os-release")
+	}
+	if err != nil {
+		return "", fmt.Errorf("error opening file : %v", err)
+	}
+	line := rex.FindAllStringSubmatch(string(bytes), -1)
+	if len(line) > 0 {
+		return strings.Trim(line[0][2], "\""), nil
+	}
+	return "Linux", nil
```

Comment:
Are we sure to default it with "Linux" rather than "Unknown" or return a blank string?

Reply:
Makes sense as we put linux as operating system in other field. Changed it to unknown